Mathlib/Analysis/Calculus/FDeriv/Prod.lean

```diff
@@ -352,7 +352,7 @@ theorem hasStrictFDerivAt_apply (i : ι) (f : ∀ i, F' i) :
   let id' := ContinuousLinearMap.id 𝕜 (∀ i, F' i)
   have h := ((hasStrictFDerivAt_pi'
              (Φ := fun (f : ∀ i, F' i) (i' : ι) => f i') (Φ' := id') (x := f))).1
-  have h' : comp (proj i) id' = proj i := by rfl
+  have h' : comp (proj i) id' = proj i := by ext; simp [id']
   rw [← h']; apply h; apply hasStrictFDerivAt_id
```

Comment:
Not sure this is an improvement. I'll probably do
```
  rw [show proj i = comp (proj i) id' from rfl]; apply h; apply hasStrictFDerivAt_id
```

Reply:
I think that's less readable than either the existing code or the code I propose

Reply:
I agree that `rw [show ...]` is not usually preferable to `have... ; rw`